extensions/vscode/vscode.wake

```diff
@@ -78,39 +69,32 @@ export def vscode _: Result Path Error =
         | runJob
         | getJobOutputs
 
-    require Pass stdlibSources =
-        sources "share/wake/lib" `.*`
+    require Pass stdlibSources = sources "share/wake/lib" `.*`
+
     require Pass stdlib =
         stdlibSources
         | findFailFn installRO
-    require Pass stdlibJSON =
-        createStdlibJSON  stdlibSources
-
-    require Pass currentDirSources =
-        sources here `[^/]*`
-    require Pass imagesSources =
-        sources here `images/.*`
-    require Pass clientSources =
-        sources here `lsp-client/.*`
-    require Pass serverSources =
-        sources here `lsp-server/.*`
-    require Pass syntaxSources =
-        sources here `syntaxes/.*`
-    require Pass vsceStdin =
-        source "{@here}/vsce-stdin"
+
+    require Pass stdlibJSON = createStdlibJSON stdlibSources
+    require Pass currentDirSources = sources @here `[^/]*`
+    require Pass imagesSources = sources @here `images/.*`
+    require Pass clientSources = sources @here `lsp-client/.*`
+    require Pass serverSources = sources @here `lsp-server/.*`
+    require Pass syntaxSources = sources @here `syntaxes/.*`
+    require Pass vsceStdin = source "{@here}/vsce-stdin"
 
     def vsceFiles =
         vsceStdin,
         jsFile,
         packageJSON,
         stdlibJSON,
-        currentDirSources ++
-        imagesSources ++
-        clientSources ++
-        serverSources ++
-        syntaxSources ++
-        stdlib ++
-        nodeModules
+        currentDirSources
+        ++ imagesSources
+        ++ clientSources
+        ++ serverSources
+        ++ syntaxSources
+        ++ stdlib
+        ++ nodeModules
```

Comment:
We do it the other way for commas, should be easy enough to add a case for `++` as well (though its different because we want a space in there unlike in the comma case)

Reply:
So this _should_ have been as simple as adding `case TOKEN_OP_ADDSUB: return true` to  `is_op_suffix`, but something weird is happening with another system that doesn't play well together.

We support formatting the code below

```
def x = (
  x # comment
  + 5
)
```

but with `+` (`+` and `++` are the same token) now marked as a suffix it falls over pretty hard

```
def x =
    (
        x # comment
+
        5
    )
```

aside: this is actually also a problem with `,` but no one would ever write

```
def t8 = (
  x # comment
  , Nil
)
```

not really sure what to do here atm. Feel free to pitch ideas otherwise I'll have to mull on it for a bit

Reply:
wait what? Do you know why that's happening?

Reply:
> aside: this is actually also a problem with , but no one would ever write
> 
> ```wake
> def t8 = (
>   x # comment
>   , Nil
> )
> ```

It's how I write my Haskell! 😉  I don't try to do it in Wake, but there's a significant minority there who like being able to simply read down the left column:

```haskell
t8 =
    [ x
    , y
    ]
record =
    { a = x
    , b = y
    }
```

But, yeah, very weird bug.  It's probably worth spending a bit of time tracking down for the MVP on its own merit, since it seems like something which *could* be innocently used even if it's not expected people do, but which breaks the code to an outsized degree.

Reply:
I have a vague idea. Basically the binop subsystem tries to be smart on how it handles the multiplicative explosion of  comment placement and operators. I think its missing a case for this specific combo (suffix op + prefix comment) because it was just so uncommon before. its also unclear to me _how_ that should even be formatted?

```
def t8 = (
  x # comment
  ,
  Nil
)
```
is the most consistent way and I certainly don't want to make some weird exception like "suffix ops become prefix ops if preceded by a comment", but its also not the prettiest. Especially in the addition case

```
def t8 = (
  x # comment
  +
  5
)
```

Reply:
I think in those cases we continue not newlining but do space.  I agree it makes suffix operators a bit weird, but I think if we suffix *all* binary operators by default (with only a few specific exceptions) rather than dividing them into two behavioural classes it's less surprising.
```
def t8 = (
  x # comment
  , Nil
)
```
```
def t8 = (
  x # comment
  + 5
)
```
Maybe the way to describe it in the formatter DSL would be something like "if the left operand ends in a comment (or if the operator is prefixed by a comment?) then print `LHS<comment><newline>OP<space>RHS`; otherwise if the operator is in the `,`/`|`/`$` classes then handle those as special cases; otherwise print `LHS<space>OP<newline or space>RHS`".  Hopefully that logic works with the structures involved, though given you've got hesitations I gather it's not quite so simple.